### PR TITLE
Add proper long line wrapping.

### PR DESCRIPTION
**Pull request by apeiron**
When sending a message to IRC, the message is relayed to other
servers and other clients by the server and needs to fit within
the max line length for server messages (512). What this means
is that the actual content of the message needs to be truncated
based on the length of the user's nick, hostname, and the kind
of message being sent (PRIVMSG, NOTICE). The current code,
in addition to hardcoding the wrap length at the absolute
max line length (not taking any of this into consideration),
is rather ugly and hard to follow. I've abstracted the logic
into something easier to follow, correct, and reuseable. It'll
be easy to expand it to ACTION.

TODO: maintain colors and other formatting across wrapped
messages. Textual does this for PRIVMSGs, at least, so we can try
looking there for inspiration.

### CODE REVIEW

**Comment by apeiron** [comment]
I will note that this is similar to #142, but the line wrapping implementation itself is a lot cleaner. I don't see where that code actually *sets* `self.messageLength`, though it makes reference to doing so if the server specifies it. So, really, there's no difference between my pull and that one besides the fact that (IMO) my code is a lot eaier to follow.

**Comment by jirwin** [comment]
+1
